Replacing RSASSA-PKCS1 with PSS

angelaizg opened the pull request:
#### e7ec278183bc30360249760160086dabbd9a832f
<pre>
Replacing RSASSA-PKCS1 with PSS
https://bugs.webkit.org/show_bug.cgi?id=247864
rdar://102295167

Reviewed by NOBODY (OOPS!).

</pre><!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/d2741710a770c7c33ca94a0f97faa5e03700d997

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [❌ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/100806 "2 style errors") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/9950 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/33848 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/110104 "Built successfully") | [✅ 🛠 🧪 win](https://ews-build.webkit.org/#/builders/10/builds/170380 "Built successfully and passed tests") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/10887 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/85/builds/732 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/93209 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/107951 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/106587 "Passed tests") | [❌ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/78/builds/8227 "Found 3 new test failures: crypto/subtle/rsa-generate-key-malformed-parameters.html, crypto/subtle/rsa-import-key-malformed-parameters.html, imported/w3c/web-platform-tests/webrtc/idlharness.https.window.html (failure)") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/91481 "Passed tests") | [❌ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/34838 "Found 3 new test failures: crypto/subtle/rsa-generate-key-malformed-parameters.html, crypto/subtle/rsa-import-key-malformed-parameters.html, imported/w3c/web-platform-tests/WebCryptoAPI/wrapKey_unwrapKey/wrapKey_unwrapKey.https.any.worker.html (failure)") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/90136 "Passed tests") | [❌ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/22875 "Found 2 new test failures: crypto/subtle/rsa-generate-key-malformed-parameters.html, crypto/subtle/rsa-import-key-malformed-parameters.html (failure)") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/77814 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/3643 "Built successfully") | [❌ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/24398 "Found 3 new test failures: crypto/subtle/rsa-generate-key-malformed-parameters.html, crypto/subtle/rsa-import-key-malformed-parameters.html, imported/w3c/web-platform-tests/webrtc/idlharness.https.window.html (failure)") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/3665 "Built successfully") | [❌ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/84/builds/701 "Found 3 new test failures: crypto/subtle/rsa-generate-key-malformed-parameters.html, crypto/subtle/rsa-import-key-malformed-parameters.html, imported/w3c/web-platform-tests/webrtc/idlharness.https.window.html (failure)") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/9776 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/43897 "Passed tests") | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/5443 "Built successfully") | | | 
| | | | | 
<!--EWS-Status-Bubble-End-->

aproskuryakov commented on Source/WebCore/crypto/CryptoAlgorithmIdentifier.h:
```diff
@@ -30,8 +30,7 @@
 enum class CryptoAlgorithmIdentifier {
-    RSASSA_PKCS1_v1_5 = 1,
-    RSA_PSS,
+    RSA_PSS = 1,
```
Do these identifiers go into some persistent serialization? We probably cannot reuse numbers if so.

Commit:
Replacing RSASSA-PKCS1 with PSS
https://bugs.webkit.org/show_bug.cgi?id=247864
rdar://102295167

Reviewed by NOBODY (OOPS!).

Removing RSAES_v1_5 from the list of algorithms we have.

* LayoutTests/crypto/subtle/rsaes-pkcs1-v1_5-generate-export-key-jwk-expected.txt: Removed.
* LayoutTests/crypto/subtle/rsaes-pkcs1-v1_5-generate-export-key-jwk.html: Removed.
* LayoutTests/crypto/subtle/rsaes-pkcs1-v1_5-generate-export-key-pkcs8-expected.txt: Removed.
* LayoutTests/crypto/subtle/rsaes-pkcs1-v1_5-generate-export-key-pkcs8.html: Removed.
* LayoutTests/crypto/subtle/rsaes-pkcs1-v1_5-generate-export-key-spki-expected.txt: Removed.
* LayoutTests/crypto/subtle/rsaes-pkcs1-v1_5-generate-export-key-spki.html: Removed.
* LayoutTests/crypto/subtle/rsaes-pkcs1-v1_5-generate-key-encrypt-decrypt-expected.txt: Removed.
* LayoutTests/crypto/subtle/rsaes-pkcs1-v1_5-generate-key-encrypt-decrypt.html: Removed.
* LayoutTests/crypto/subtle/rsaes-pkcs1-v1_5-generate-key-expected.txt: Removed.
* LayoutTests/crypto/subtle/rsaes-pkcs1-v1_5-generate-key-extractable-expected.txt: Removed.
* LayoutTests/crypto/subtle/rsaes-pkcs1-v1_5-generate-key-extractable.html: Removed.
* LayoutTests/crypto/subtle/rsaes-pkcs1-v1_5-generate-key.html: Removed.
* LayoutTests/crypto/subtle/rsaes-pkcs1-v1_5-import-jwk-private-key-expected.txt: Removed.
* LayoutTests/crypto/subtle/rsaes-pkcs1-v1_5-import-jwk-private-key.html: Removed.
* LayoutTests/crypto/subtle/rsaes-pkcs1-v1_5-import-jwk-public-key-empty-usages-expected.txt: Removed.
* LayoutTests/crypto/subtle/rsaes-pkcs1-v1_5-import-jwk-public-key-empty-usages.html: Removed.
* LayoutTests/crypto/subtle/rsaes-pkcs1-v1_5-import-jwk-public-key-expected.txt: Removed.
* LayoutTests/crypto/subtle/rsaes-pkcs1-v1_5-import-jwk-public-key-leading-zero-expected.txt: Removed.
* LayoutTests/crypto/subtle/rsaes-pkcs1-v1_5-import-jwk-public-key-leading-zero.html: Removed.
* LayoutTests/crypto/subtle/rsaes-pkcs1-v1_5-import-jwk-public-key-minimum-expected.txt: Removed.
* LayoutTests/crypto/subtle/rsaes-pkcs1-v1_5-import-jwk-public-key-minimum.html: Removed.
* LayoutTests/crypto/subtle/rsaes-pkcs1-v1_5-import-jwk-public-key-non-extractable-expected.txt: Removed.
* LayoutTests/crypto/subtle/rsaes-pkcs1-v1_5-import-jwk-public-key-non-extractable.html: Removed.
* LayoutTests/crypto/subtle/rsaes-pkcs1-v1_5-import-jwk-public-key.html: Removed.
* LayoutTests/crypto/subtle/rsaes-pkcs1-v1_5-import-key-decrypt-expected.txt: Removed.
* LayoutTests/crypto/subtle/rsaes-pkcs1-v1_5-import-key-decrypt.html: Removed.
* LayoutTests/crypto/subtle/rsaes-pkcs1-v1_5-import-key-encrypt-expected.txt: Removed.
* LayoutTests/crypto/subtle/rsaes-pkcs1-v1_5-import-key-encrypt.html: Removed.
* LayoutTests/crypto/subtle/rsaes-pkcs1-v1_5-import-pkcs8-key-expected.txt: Removed.
* LayoutTests/crypto/subtle/rsaes-pkcs1-v1_5-import-pkcs8-key.html: Removed.
* LayoutTests/crypto/subtle/rsaes-pkcs1-v1_5-import-spki-key-empty-usages-expected.txt: Removed.
* LayoutTests/crypto/subtle/rsaes-pkcs1-v1_5-import-spki-key-empty-usages.html: Removed.
* LayoutTests/crypto/subtle/rsaes-pkcs1-v1_5-import-spki-key-expected.txt: Removed.
* LayoutTests/crypto/subtle/rsaes-pkcs1-v1_5-import-spki-key.html: Removed.
* Source/WebCore/Sources.txt:
* Source/WebCore/SourcesCocoa.txt:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/bindings/js/SerializedScriptValue.cpp:
(WebCore::CloneSerializer::write):
(WebCore::CloneDeserializer::read):
* Source/WebCore/crypto/CryptoAlgorithmIdentifier.h:
* Source/WebCore/crypto/SubtleCrypto.cpp:
(WebCore::normalizeCryptoAlgorithmParameters):
(WebCore::isSupportedExportKey):
* Source/WebCore/crypto/gcrypt/CryptoAlgorithmRSAES_PKCS1_v1_5GCrypt.cpp: Removed.
* Source/WebCore/crypto/gcrypt/CryptoAlgorithmRegistryGCrypt.cpp:
(WebCore::CryptoAlgorithmRegistry::platformRegisterAlgorithms):
* Source/WebCore/crypto/mac/CryptoAlgorithmAES_CTRMac.cpp:
* Source/WebCore/crypto/mac/CryptoAlgorithmRegistryMac.cpp:
(WebCore::CryptoAlgorithmRegistry::platformRegisterAlgorithms):
* Source/WebCore/crypto/openssl/CryptoAlgorithmRSAES_PKCS1_v1_5OpenSSL.cpp: Removed.
* Source/WebCore/crypto/openssl/CryptoAlgorithmRegistryOpenSSL.cpp:
(WebCore::CryptoAlgorithmRegistry::platformRegisterAlgorithms):

Remove insecure Crypto algorithms from WebCrypto
https://bugs.webkit.org/show_bug.cgi?id=247358
rdar://101765059

Reviewed by NOBODY (OOPS!).

Removing RSAES_v1_5 from the list of algorithms we have.

* LayoutTests/crypto/subtle/rsaes-pkcs1-v1_5-generate-export-key-jwk-expected.txt: Removed.
* LayoutTests/crypto/subtle/rsaes-pkcs1-v1_5-generate-export-key-jwk.html: Removed.
* LayoutTests/crypto/subtle/rsaes-pkcs1-v1_5-generate-export-key-pkcs8-expected.txt: Removed.
* LayoutTests/crypto/subtle/rsaes-pkcs1-v1_5-generate-export-key-pkcs8.html: Removed.
* LayoutTests/crypto/subtle/rsaes-pkcs1-v1_5-generate-export-key-spki-expected.txt: Removed.
* LayoutTests/crypto/subtle/rsaes-pkcs1-v1_5-generate-export-key-spki.html: Removed.
* LayoutTests/crypto/subtle/rsaes-pkcs1-v1_5-generate-key-encrypt-decrypt-expected.txt: Removed.
* LayoutTests/crypto/subtle/rsaes-pkcs1-v1_5-generate-key-encrypt-decrypt.html: Removed.
* LayoutTests/crypto/subtle/rsaes-pkcs1-v1_5-generate-key-expected.txt: Removed.
* LayoutTests/crypto/subtle/rsaes-pkcs1-v1_5-generate-key-extractable-expected.txt: Removed.
* LayoutTests/crypto/subtle/rsaes-pkcs1-v1_5-generate-key-extractable.html: Removed.
* LayoutTests/crypto/subtle/rsaes-pkcs1-v1_5-generate-key.html: Removed.
* LayoutTests/crypto/subtle/rsaes-pkcs1-v1_5-import-jwk-private-key-expected.txt: Removed.
* LayoutTests/crypto/subtle/rsaes-pkcs1-v1_5-import-jwk-private-key.html: Removed.
* LayoutTests/crypto/subtle/rsaes-pkcs1-v1_5-import-jwk-public-key-empty-usages-expected.txt: Removed.
* LayoutTests/crypto/subtle/rsaes-pkcs1-v1_5-import-jwk-public-key-empty-usages.html: Removed.
* LayoutTests/crypto/subtle/rsaes-pkcs1-v1_5-import-jwk-public-key-expected.txt: Removed.
* LayoutTests/crypto/subtle/rsaes-pkcs1-v1_5-import-jwk-public-key-leading-zero-expected.txt: Removed.
* LayoutTests/crypto/subtle/rsaes-pkcs1-v1_5-import-jwk-public-key-leading-zero.html: Removed.
* LayoutTests/crypto/subtle/rsaes-pkcs1-v1_5-import-jwk-public-key-minimum-expected.txt: Removed.
* LayoutTests/crypto/subtle/rsaes-pkcs1-v1_5-import-jwk-public-key-minimum.html: Removed.
* LayoutTests/crypto/subtle/rsaes-pkcs1-v1_5-import-jwk-public-key-non-extractable-expected.txt: Removed.
* LayoutTests/crypto/subtle/rsaes-pkcs1-v1_5-import-jwk-public-key-non-extractable.html: Removed.
* LayoutTests/crypto/subtle/rsaes-pkcs1-v1_5-import-jwk-public-key.html: Removed.
* LayoutTests/crypto/subtle/rsaes-pkcs1-v1_5-import-key-decrypt-expected.txt: Removed.
* LayoutTests/crypto/subtle/rsaes-pkcs1-v1_5-import-key-decrypt.html: Removed.
* LayoutTests/crypto/subtle/rsaes-pkcs1-v1_5-import-key-encrypt-expected.txt: Removed.
* LayoutTests/crypto/subtle/rsaes-pkcs1-v1_5-import-key-encrypt.html: Removed.
* LayoutTests/crypto/subtle/rsaes-pkcs1-v1_5-import-pkcs8-key-expected.txt: Removed.
* LayoutTests/crypto/subtle/rsaes-pkcs1-v1_5-import-pkcs8-key.html: Removed.
* LayoutTests/crypto/subtle/rsaes-pkcs1-v1_5-import-spki-key-empty-usages-expected.txt: Removed.
* LayoutTests/crypto/subtle/rsaes-pkcs1-v1_5-import-spki-key-empty-usages.html: Removed.
* LayoutTests/crypto/subtle/rsaes-pkcs1-v1_5-import-spki-key-expected.txt: Removed.
* LayoutTests/crypto/subtle/rsaes-pkcs1-v1_5-import-spki-key.html: Removed.
* Source/WebCore/Sources.txt:
* Source/WebCore/SourcesCocoa.txt:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/bindings/js/SerializedScriptValue.cpp:
(WebCore::CloneSerializer::write):
(WebCore::CloneDeserializer::read):
* Source/WebCore/crypto/CryptoAlgorithmIdentifier.h:
* Source/WebCore/crypto/SubtleCrypto.cpp:
(WebCore::normalizeCryptoAlgorithmParameters):
(WebCore::isSupportedExportKey):
* Source/WebCore/crypto/gcrypt/CryptoAlgorithmRSAES_PKCS1_v1_5GCrypt.cpp: Removed.
* Source/WebCore/crypto/gcrypt/CryptoAlgorithmRegistryGCrypt.cpp:
(WebCore::CryptoAlgorithmRegistry::platformRegisterAlgorithms):
* Source/WebCore/crypto/mac/CryptoAlgorithmAES_CTRMac.cpp:
* Source/WebCore/crypto/mac/CryptoAlgorithmRegistryMac.cpp:
(WebCore::CryptoAlgorithmRegistry::platformRegisterAlgorithms):
* Source/WebCore/crypto/openssl/CryptoAlgorithmRSAES_PKCS1_v1_5OpenSSL.cpp: Removed.
* Source/WebCore/crypto/openssl/CryptoAlgorithmRegistryOpenSSL.cpp:
(WebCore::CryptoAlgorithmRegistry::platformRegisterAlgorithms):

Need a short description (OOPS!).
Need the bug URL (OOPS!).
Include a Radar link (OOPS!).

Reviewed by NOBODY (OOPS!).

* Source/WebCore/Modules/mediastream/PeerConnectionBackend.h:
(WebCore::PeerConnectionBackend::CertificateInformation::RSASS_PSS):
(WebCore::PeerConnectionBackend::CertificateInformation::RSASSA_PKCS1_v1_5): Deleted.
* Source/WebCore/Modules/mediastream/RTCPeerConnection.cpp:
(WebCore::certificateTypeFromAlgorithmIdentifier):

Replacing RSASSA-PKCS1 with PSS
https://bugs.webkit.org/show_bug.cgi?id=247864
rdar://102295167

Reviewed by NOBODY (OOPS!).

* Source/WebCore/Modules/mediastream/RTCPeerConnection.cpp:
(WebCore::certificateTypeFromAlgorithmIdentifier):
* Source/WebCore/Sources.txt:
* Source/WebCore/SourcesCocoa.txt:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/bindings/js/SerializedScriptValue.cpp:
(WebCore::CloneSerializer::write):
(WebCore::CloneDeserializer::read):
* Source/WebCore/crypto/CryptoAlgorithmIdentifier.h:
* Source/WebCore/crypto/SubtleCrypto.cpp:
(WebCore::normalizeCryptoAlgorithmParameters):
(WebCore::isSupportedExportKey):
* Source/WebCore/crypto/algorithms/CryptoAlgorithmRSASSA_PKCS1_v1_5.cpp: Removed.
* Source/WebCore/crypto/gcrypt/CryptoAlgorithmRSASSA_PKCS1_v1_5GCrypt.cpp: Removed.
* Source/WebCore/crypto/gcrypt/CryptoAlgorithmRegistryGCrypt.cpp:
(WebCore::CryptoAlgorithmRegistry::platformRegisterAlgorithms):
* Source/WebCore/crypto/mac/CryptoAlgorithmRSASSA_PKCS1_v1_5Mac.cpp: Removed.
* Source/WebCore/crypto/mac/CryptoAlgorithmRegistryMac.cpp:
(WebCore::CryptoAlgorithmRegistry::platformRegisterAlgorithms):
* Source/WebCore/crypto/openssl/CryptoAlgorithmRSASSA_PKCS1_v1_5OpenSSL.cpp: Removed.
* Source/WebCore/crypto/openssl/CryptoAlgorithmRegistryOpenSSL.cpp:
(WebCore::CryptoAlgorithmRegistry::platformRegisterAlgorithms):
* Source/WebCore/platform/OpenSSL.cmake:
* Tools/TestWebKitAPI/Tests/WebKit/navigation-client-default-crypto.html:

Replacing RSASSA-PKCS1 with PSS
https://bugs.webkit.org/show_bug.cgi?id=247864
rdar://102295167

Reviewed by NOBODY (OOPS!).

* Source/WebCore/Modules/mediastream/RTCPeerConnection.cpp:
(WebCore::certificateTypeFromAlgorithmIdentifier):
* Source/WebCore/Sources.txt:
* Source/WebCore/SourcesCocoa.txt:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/bindings/js/SerializedScriptValue.cpp:
(WebCore::CloneSerializer::write):
(WebCore::CloneDeserializer::read):
* Source/WebCore/crypto/CryptoAlgorithmIdentifier.h:
* Source/WebCore/crypto/SubtleCrypto.cpp:
(WebCore::normalizeCryptoAlgorithmParameters):
(WebCore::isSupportedExportKey):
* Source/WebCore/crypto/algorithms/CryptoAlgorithmRSASSA_PKCS1_v1_5.cpp: Removed.
* Source/WebCore/crypto/gcrypt/CryptoAlgorithmRSASSA_PKCS1_v1_5GCrypt.cpp: Removed.
* Source/WebCore/crypto/gcrypt/CryptoAlgorithmRegistryGCrypt.cpp:
(WebCore::CryptoAlgorithmRegistry::platformRegisterAlgorithms):
* Source/WebCore/crypto/mac/CryptoAlgorithmRSASSA_PKCS1_v1_5Mac.cpp: Removed.
* Source/WebCore/crypto/mac/CryptoAlgorithmRegistryMac.cpp:
(WebCore::CryptoAlgorithmRegistry::platformRegisterAlgorithms):
* Source/WebCore/crypto/openssl/CryptoAlgorithmRSASSA_PKCS1_v1_5OpenSSL.cpp: Removed.
* Source/WebCore/crypto/openssl/CryptoAlgorithmRegistryOpenSSL.cpp:
(WebCore::CryptoAlgorithmRegistry::platformRegisterAlgorithms):
* Source/WebCore/platform/OpenSSL.cmake:
* Tools/TestWebKitAPI/Tests/WebKit/navigation-client-default-crypto.html:

Replacing RSASSA-PKCS1 with PSS
https://bugs.webkit.org/show_bug.cgi?id=247864
rdar://102295167

Reviewed by NOBODY (OOPS!).

* Tools/TestWebKitAPI/Tests/WebKit/navigation-client-default-crypto.html:

Replacing RSASSA-PKCS1 with PSS
https://bugs.webkit.org/show_bug.cgi?id=247864
rdar://102295167

Reviewed by NOBODY (OOPS!).

* Tools/TestWebKitAPI/Tests/WebKit/navigation-client-default-crypto.html:
